numpy/_core/tests/test_numeric.py

```diff
@@ -3596,6 +3596,25 @@ def test_mode(self):
         with assert_raises(TypeError):
             np.correlate(d, k, mode=None)
 
+    def test_out(self):
+        self._setup(float)
+        out = np.zeros(self.z1.shape)
+        res = np.correlate(self.x, self.y, 'full', out=out)
+        assert_equal(res, out)
+        assert_array_almost_equal(out, self.z1)
+        out = np.zeros(self.z2.shape)
+        res = np.correlate(self.y, self.x, 'full', out=out)
+        assert_equal(res, out)
+        assert_array_almost_equal(out, self.z2)
+        out = np.zeros(self.z1r.shape)
+        res = np.correlate(self.x[::-1], self.y, 'full', out=out)
+        assert_equal(res, out)
+        assert_array_almost_equal(out, self.z1r)
+        out = np.zeros(self.z2r.shape)
+        res = np.correlate(self.y, self.x[::-1], 'full', out=out)
+        assert_equal(res, out)
+        assert_array_almost_equal(out, self.z2r)
```

Comment:
Can you add blank lines and split it up into multiple tests (or parametrization) if it makes sense?
(Minor improvement, this is OK overall).